spec/core/ics-002-client-semantics/README.md

```diff
@@ -660,29 +569,43 @@ to allow governance mechanisms to perform these actions
 ```typescript
 function updateClient(
   id: Identifier,
-  header: Header) {
-    clientType = provableStore.get(clientTypePath(id))
-    abortTransactionUnless(clientType !== null)
-    clientState = provableStore.get(clientStatePath(id))
-    abortTransactionUnless(clientState !== null)
-    clientType.checkValidityAndUpdateState(clientState, header)
+  clientMessage: ClientMessage) {
+    if err := clientState.VerifyClientMessage(clientMessage); err != nil {
+        return err
+    }
+    
+    foundMisbehaviour := clientState.CheckForMisbehaviour(clientMessage)
+    if foundMisbehaviour {
+        clientState.UpdateStateOnMisbehaviour(header)
+        // emit misbehaviour event
+        return 
+    }
+    
+    clientState.UpdateState(clientMessage) // expects no-op on duplicate header
+    // emit update event
+    return
 }
 ```
 
 #### Misbehaviour
 
-If the client detects proof of misbehaviour, the client can be alerted, possibly invalidating
+If a relayer wishes to explicitly submit misbehaviour. They may alert the client to the misbehaviour directly, possibly invalidating
 previously valid state roots & preventing future updates.
 
 ```typescript
 function submitMisbehaviourToClient(
```

Comment:
Why do we still need `submitMisbehaviourToClient` given the new meaning of `updateClient`?

Reply:
`submitMisbehaviourToClient` is for explicit misbehaviour handling. THe relayer knows the message it is sending will shutdown the client

In UpdateClient, it is possible for innocent attempts to add a new header to actually be proof of misbehaviour which we catch and handle.

In the future we could possibly deprecate `submitMisbehaviourToClient` but we won't for now to minimize disruptions to relayers and light client implementers